src/main/kotlin/io/github/mojira/arisa/ModuleExecutor.kt

```diff
@@ -0,0 +1,304 @@
+                )
+            )
+        }
+        exec(Arisa.Modules.ReopenAwaiting) {
+            "ReopenAwaiting" to reopenAwaitingModule(
+                ReopenAwaitingModuleRequest(
+                    it,
+                    it.resolution,
+                    (it.getField("created") as String).toInstant(),
+                    (it.getField("updated") as String).toInstant(),
+                    it.comments
+                )
+            )
+        }
+        exec(Arisa.Modules.RevokeConfirmation) {
+            "RevokeConfirmation" to revokeConfirmationModule(
+                RevokeConfirmationModuleRequest(
+                    it,
+                    ((it.getField(config[Arisa.CustomFields.confirmationField])) as? JSONObject)?.get("value") as? String?,
+                    it.changeLog.entries
+                )
+            )
+        }
+
+        processedTickets
+            .filter { (_, executed) -> !executed }
+            .map { (ticket, _) -> ticket }
+            .forEach { ticket ->
+                ticketCache.add(ticket)
+                ticketTimer.schedule(290_000) { ticketCache.remove(ticket) }
+            }
+    }
+
+    private fun executeModule(
+        moduleConfig: Arisa.Modules.ModuleConfigSpec,
+        queryCache: MutableMap<String, List<Issue>>,
+        processedIssues: MutableMap<String, Boolean>,
+        executeModule: (Issue) -> Pair<String, Either<ModuleError, ModuleResponse>>
+    ) {
+        val projects = config[Arisa.Issues.projects].joinToString(",")
+        val cachedTickets = ticketCache.joinToString(",")
+        val combinedJql = "project in ($projects) AND key not in ($cachedTickets) AND (${config[moduleConfig.jql]})"
+
+        val issues = queryCache[combinedJql] ?: jiraClient
+            .searchIssues(combinedJql)
+            .issues
+            .map { jiraClient.getIssue(it.key, "*all", "changelog") } // Get issues again to retrieve all fields
+            .filter { issue ->
+                // Ignore issues where last action was a resolve
+                val latestChange = issue.changeLog.entries.lastOrNull()
+
+                latestChange == null || // There is actually a entry
+                    !latestChange.items.any { it.field == "resolution" } || // It was a transition
+                    latestChange.author.name == config[Arisa.Credentials.username] || // The transition was not done by the bot
+                    (issue.comments.isNotEmpty() && issue.comments.last().updatedDate > latestChange.created) // And there is no comment posted after that
+            }
+        queryCache[combinedJql] = issues
+
+        issues
+            .filter { config[moduleConfig.whitelist] == null || config[moduleConfig.whitelist]!!.contains(it.key) }
+            .filter { config[moduleConfig.resolutions].contains(it.resolution.name) }
+            .map { it.key to executeModule(it) }
+            .forEach { (issue, response) ->
+                when (response.second) {
```

Comment:
Can be response.second.fold(
  {//left part},
  {//right part}
)

That also automatically casts stuff so you dont need the with